### PR TITLE
Fix typos in the networkx codebase

### PR DESCRIPTION
**Pull request by faze-geek**
I could see some issues related to fixing typos, hence I ran the `codespell` library over the `networkx` codebase and could fix various typos.
I have personally checked each typo to avoid any false positives.

### CODE REVIEW

**Comment by faze-geek** [comment]
Not sure why checks have not started !
Kindly help me with that.

File: networkx/algorithms/operators/product.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def _dict_product(d1, d2):
     return {k: (d1.get(k), d2.get(k)) for k in set(d1) | set(d2)}
 
 
-# Generators for producting graph products
+# Generators for producing graph products
```
Comment (MridulS) [inline networkx/algorithms/operators/product.py:24]: Maybe this isn't a typo. We are indeed "producting" graph products.

File: networkx/readwrite/gml.py [hunk 2]
```diff
@@ -372,7 +372,7 @@ def parse_kv(curr_token):
             elif category == Pattern.DICT_START:
                 curr_token, value = parse_dict(curr_token)
             else:
-                # Allow for string convertible id and label values
+                # Allow for string convertable id and label values
```
Comment (MridulS) [inline networkx/readwrite/gml.py:375]: Wait isn't `convertible` the right spelling?

**Review by MridulS — APPROVED** [review]
Well we do make a bunch of typos.

**Comment by faze-geek** [comment]
Thanks @MridulS for the comments . I'll make the changes quick.

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**